include/triton/Dialect/TritonGPU/IR/TritonGPUOps.td

```diff
@@ -239,35 +159,63 @@ def TTG_InsertSliceAsyncOp : TTG_Op<"insert_slice_async",
     }
   }];
 
-  let hasCustomAssemblyFormat = 1;
+  let assemblyFormat = [{
+    $src `,` $dst (`mask` $mask^)? (`other` $other^)? attr-dict `:` type($src) `->` type($dst)
+  }];
 }
 
-def TTG_AllocTensorOp : TTG_Op<"alloc_tensor", [MemoryEffects<[MemAlloc<SharedMemory>]>,  // Allocate shared memory
-                                                ResultsAreSharedEncoding]> {
+
+// Allocate shared memory
+def TTG_LocalAllocOp : TTG_Op<"local_alloc", [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
   let summary = "allocate tensor";
+  let description = [{
+    This operation allocates buffer in shared memory and return a descriptor
+    containing the address and a view of the buffer.
+  }];
+  let arguments = (ins Optional<TT_Tensor>:$init);
+
+  let assemblyFormat = [{$init attr-dict `:` functional-type(operands, results)}];
+
+  let results = (outs TT_MemDescType:$result);
+}
+
+// Deallocate shared memory
+def TTG_DeallocOp : TTG_Op<"dealloc", [MemoryEffects<[MemFree<SharedMemory>]>]> {
```

Comment:
Wondering why there is no "local" in the name of this op, compared to `LocalAllocOp`.

Reply:
good point, changing the name.